src/provider/model/AppDirectory.ts

```diff
@@ -16,6 +16,44 @@ enum StorageKeys {
 
 @injectable()
 export class AppDirectory extends AsyncInit {
+    /**
+     * Test if an app *might* support an intent - i.e., were the app running with the appropriate listener added,
+     * should we regard it as supporting this intent (and optionally context)
+     */
+    public static mightAppSupportIntent(app: Application, intentType: string, contextType?: string): boolean {
```

Comment:
Open to better naming on `mightAppSupportIntent` and `shouldAppSupportIntent`, as current naming makes test descriptions/titles awkward.

Tempted by `hopeAppSupportsIntent` and `expectAppSupportsIntent`...